syntax = "proto3";

package services.caption_service.v1;

import "google/api/annotations.proto";
import "messages/image_dto/v1/image_dto.proto";

// Request message for GetHashtags
message GetHashtagsRequest {
	// Caption for which hashtags are to be generated
	string caption = 1;
}

// Response message for GetHashtags
message GetHashtagsResponse {
	// List of hashtags generated for the caption
	string hashtags_csv = 1;
}

// Service for all hashtag related operations
service HashtagService {
	// Get hashtags for a give caption
	rpc GetHashtags(GetHashtagsRequest) returns (GetHashtagsResponse) {
		option (google.api.http) = {
			get: "/v1/hashtags"
		};
	}}
	
   
